test/follow.test.js

```diff
@@ -0,0 +1,153 @@
+        const response = await chai
+          .request(server)
+          .post('/api/v1/profiles/marydoe/follow');
+        expect(response.body.error).to.equal('No token provided');
+      });
+
+    it('Should not be able to follow user with invalid token provided',
+      async () => {
+        const response = await chai
+          .request(server)
+          .post('/api/v1/profiles/marydoe/follow')
+          .set('Authorization', 'someInvalidToken');
+        expect(response.body.error).to.equal('Invalid token');
+      });
+
+    it('Should not be able to follow a non-existent user',
+      async () => {
+        const response = await chai
+          .request(server)
+          .post('/api/v1/profiles/johndoe/follow')
+          .set('Authorization', firstUserToken);
+        expect(response.body.error)
+          .to.equal('The user provided does not exist');
+      });
+
+    it('Should not be able to follow self',
+      async () => {
+        const response = await chai
+          .request(server)
+          .post('/api/v1/profiles/johnsolomon/follow')
+          .set('Authorization', firstUserToken);
+        expect(response.body.error).to.equal('You cannot follow yourself');
+      });
+
+    it('Should be able to follow another user',
+      async () => {
+        const response = await chai
+          .request(server)
+          .post('/api/v1/profiles/marydoe/follow')
+          .set('Authorization', firstUserToken);
+        expect(response.body.message).to.equal('You are now following marydoe');
+      });
+
+    it('Should not be able to follow the same user twice',
+      async () => {
+        const response = await chai
+          .request(server)
+          .post('/api/v1/profiles/marydoe/follow')
+          .set('Authorization', firstUserToken);
+        expect(response.body.error)
+          .to.equal('You are already following this user');
+      });
+  });
+
+  describe('Followers', () => {
+    it('Should not be able to get followers when no token is provided',
+      async () => {
+        const response = await chai
+          .request(server)
+          .post('/api/v1/profile/followers');
+        expect(response.body.error).to.equal('No token provided');
+      });
+
+    it('Should not be able to get followers with invalid token provided',
+      async () => {
+        const response = await chai
+          .request(server)
+          .post('/api/v1/profile/followers')
+          .set('Authorization', 'someInvalidToken');
+        expect(response.body.error).to.equal('Invalid token');
+      });
+
+    it('Should be able to get followers with valid token',
+      async () => {
+        const response = await chai
+          .request(server)
+          .get('/api/v1/profile/followers')
+          .set('Authorization', secondUserToken);
+        expect(
+          response.body.followers[0].followerDetails.username
+        ).to.equal('johnsolomon');
+      });
+  });
+
+  describe('Following', () => {
```

Comment:
In this describe description, you can add the route here. Also, may be a good touch to make the `Following` `FOLLOWING`, so the whole thing becomes `GET A USER'S FOLLOWING LIST: /api/v1/profile/following`